src/java/main/org/apache/zookeeper/server/SessionTrackerImpl.java

```diff
@@ -280,6 +278,12 @@ public synchronized boolean addSession(long id, int sessionTimeout) {
         return added;
     }
 
+    public synchronized boolean commitSession(long id, int sessionTimeout) {
```

Comment:
`sessionsWithTimeout` is a Map within ZKdb, so adding sessions to it is equivalent to persisting them.

I think you could also remove the `addGlobalSession()` method completely, because it doesn't do anything special than forwarding the call to `addSession()`.

Reply:
Yes, that's kind of commit changes to ZkDB, like the usual txns commit to DataTree in ZkDB.